pkg/inspector/image-inspector.go

```diff
@@ -37,193 +38,37 @@ const (
+	return &defaultImageInspector{opts}
 }
 
-func processTarStream(tr *tar.Reader, destination string) error {
-	for {
-		hdr, err := tr.Next()
-		if err != nil {
-			if err == io.EOF {
-				return nil
-			}
-			return fmt.Errorf("Unable to extract container: %v\n", err)
-		}
-
-		hdrInfo := hdr.FileInfo()
-
-		dstpath := path.Join(destination, strings.TrimPrefix(hdr.Name, DOCKER_TAR_PREFIX))
-		// Overriding permissions to allow writing content
-		mode := hdrInfo.Mode() | OWNER_PERM_RW
-
-		switch hdr.Typeflag {
-		case tar.TypeDir:
-			if err := os.Mkdir(dstpath, mode); err != nil {
-				if !os.IsExist(err) {
-					return fmt.Errorf("Unable to create directory: %v", err)
-				}
-				err = os.Chmod(dstpath, mode)
-				if err != nil {
-					return fmt.Errorf("Unable to update directory mode: %v", err)
-				}
-			}
-		case tar.TypeReg, tar.TypeRegA:
-			file, err := os.OpenFile(dstpath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, mode)
-			if err != nil {
-				return fmt.Errorf("Unable to create file: %v", err)
-			}
-			if _, err := io.Copy(file, tr); err != nil {
-				file.Close()
-				return fmt.Errorf("Unable to write into file: %v", err)
-			}
-			file.Close()
-		case tar.TypeSymlink:
-			if err := os.Symlink(hdr.Linkname, dstpath); err != nil {
-				return fmt.Errorf("Unable to create symlink: %v\n", err)
-			}
-		case tar.TypeLink:
-			target := path.Join(destination, strings.TrimPrefix(hdr.Linkname, DOCKER_TAR_PREFIX))
-			if err := os.Link(target, dstpath); err != nil {
-				return fmt.Errorf("Unable to create link: %v\n", err)
-			}
-		default:
-			// For now we're skipping anything else. Special device files and
-			// symlinks are not needed or anyway probably incorrect.
-		}
-
-		// maintaining access and modification time in best effort fashion
-		os.Chtimes(dstpath, hdr.AccessTime, hdr.ModTime)
-	}
-}
-
-func generateRandomName() string {
-	n, err := rand.Int(rand.Reader, big.NewInt(math.MaxInt64))
+// Inspect inspects and serves the image based on the ImageInspectorOptions.
+func (i *defaultImageInspector) Inspect() error {
+	client, err := docker.NewClient(i.opts.URI)
 	if err != nil {
-		log.Fatalf("Unable to generate random container name: %v\n", err)
+		return fmt.Errorf("Unable to connect to docker daemon: %v\n", err)
 	}
-	return fmt.Sprintf("image-inspector-%016x", n)
-}
 
-func getAuthConfigs(dockercfg, username, password_file string) *docker.AuthConfigurations {
-	imagePullAuths := &docker.AuthConfigurations{
-		map[string]docker.AuthConfiguration{"": {}}}
-	if dockercfg != "" {
-		reader, err := os.Open(dockercfg)
-		if err != nil {
-			log.Fatalf("Unable to open docker config file: %v\n", err)
+	if _, err := client.InspectImage(i.opts.Image); err != nil {
+		log.Printf("Pulling image %s", i.opts.Image)
+		imagePullOption := docker.PullImageOptions{Repository: i.opts.Image}
+		imagePullAuths, authCfgErr := getAuthConfigs(i.opts.DockerCfg, i.opts.Username, i.opts.PasswordFile)
```

Comment:
we could but we'd have to pre-declare the `imagePullAuths` so I figured it's same/same either way (new error or new line for `imagePullAuths`).  Happy to update it though.  

For the record I didn't reuse `err` so we wouldn't shadow it in case it was ever used later in the life of this code.